File: Behavioural-Patterns/ChainOfResponsibility/Java/Implementations/AuthorizationHandler.java
package Implementations;

import Interfaces.BaseHandler;

public class AuthorizationHandler extends BaseHandler{

    @Override
    protected boolean processRequest(Request request) {
        System.out.println("-> AuthorizationHandler: Checking authorization...");

        if ("admin".equals(request.getHeader("role"))) {
            System.out.println("Authorization successful (User is an admin).");
            return true;
        } else {
            System.out.println("Authorization FAILED. User does not have admin role. Request rejected.");
            return false;
        }
    }
    
}

File: Behavioural-Patterns/ChainOfResponsibility/C#/ChainOfResponsibility/Implementations/AuthorizationHandler.cs
public class AuthorizationHandler : BaseHandler{

    protected override bool processRequest(Request request) {
        Console.WriteLine("-> AuthorizationHandler: Checking authorization...");

        if ("admin".Equals(request.getHeader("role"))) {
            Console.WriteLine("Authorization successful (User is an admin).");
            return true;
        } else {
            Console.WriteLine("Authorization FAILED. User does not have admin role. Request rejected.");
            return false;
        }
    }
    
}
